processor/spanmetricsprocessor/processor.go

```diff
@@ -350,6 +350,7 @@ func (p *processorImp) resetAccumulatedMetrics() {
 	p.latencyCount = make(map[metricKey]uint64)
 	p.latencySum = make(map[metricKey]float64)
 	p.latencyBucketCounts = make(map[metricKey][]uint64)
+	p.metricKeyToDimensions = make(map[metricKey]pdata.AttributeMap)
```

Comment:
Did this break a test case by not doing this btw?

Reply:
I can see that the tests were green, so I am wondering if there is enough testing in this to be confident  that we are doing the correct thing.

Reply:
This map is essentially a cache for mapping the metric keys to unique attribute sets. So resetting the map doesn't change any behaviour, it just ensures memory is bounded by the current collection cycle - which should be the case for delta metrics.